From a5a1861f4be644b9ca7069df886c75ccc0467229 Mon Sep 17 00:00:00 2001
From: Mathew McBride <matt@traverse.com.au>
Date: Tue, 13 Jan 2026 11:33:17 +1100
Subject: [PATCH 05/13] service_wan: use _GENERIC_LINUX_DATA_PATH_ to introduce
 generic Arm support

Similar to the previous commit for the firewall changes, introduce
_GENERIC_LINUX_DATA_PATH_ as a substitute for defined(_PLATFORM_BANANA_PI_)
where appropriate.

This will add support for the generic Arm reference platform and
any other future platforms with the same configuration.

Signed-off-by: Mathew McBride <matt@traverse.com.au>
---
 source/service_wan/service_wan.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/service_wan/service_wan.c b/source/service_wan/service_wan.c
index b9abc63..9fd4102 100644
--- a/source/service_wan/service_wan.c
+++ b/source/service_wan/service_wan.c
@@ -1508,7 +1508,7 @@ STATIC int wan_iface_up(struct serv_wan *sw)
 STATIC int wan_iface_down(struct serv_wan *sw)
 {
     int err = 0;
-#if !defined(_PLATFORM_RASPBERRYPI_)  && !defined(_PLATFORM_BANANAPI_R4_)
+#if !defined(_PLATFORM_RASPBERRYPI_)  && !defined(_GENERIC_LINUX_DATA_PATH_)
     err = v_secure_system("ip -4 link set %s down", sw->ifname);
 #endif
 #if PUMA6_OR_NEWER_SOC_TYPE
@@ -1742,7 +1742,7 @@ STATIC int wan_addr_set(struct serv_wan *sw)
     	if(strcmp(mischandler_ready,"true") == 0)
     	{
     		//only for first time
-    #if !defined(_PLATFORM_RASPBERRYPI_) && !defined(_PLATFORM_TURRIS_)  && !defined(_PLATFORM_BANANAPI_R4_)
+#if !defined(_PLATFORM_RASPBERRYPI_) && !defined(_GENERIC_LINUX_DATA_PATH_)
     		fprintf(stderr, "[%s] ready is set from misc handler. Doing gw_lan_refresh\n", PROG_NAME);
             #if defined (_XB6_PRODUCT_REQ_) && defined (_COSA_BCM_ARM_)
                 v_secure_system("firewall");
-- 
2.51.2

